libcontainer/container_setup.go

```diff
@@ -0,0 +1,149 @@
+package libcontainer
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/opencontainers/runc/libcontainer/configs"
+	"github.com/opencontainers/runc/libcontainer/system"
+	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink/nl"
+	"golang.org/x/sys/execabs"
+	"golang.org/x/sys/unix"
+)
+
+// NsExecSyncMsg is used for communication between the parent and child during
+// container setup.
+type NsExecSyncMsg uint32
+
+const (
+	syncUsermapPls NsExecSyncMsg = iota + 0x40
+	syncUsermapAck
+	syncRecvPidPls
+	syncRecvPidAck
+	syncTimeOffsetsPls
+	syncTimeOffsetsAck
```

Comment:
I see you're doing this from Go now, so these two might not be needed anymore

Reply:
It still needs here, because we should set it from the parent process after the stage-1 process requested the timeoffsets set.